.github/workflows/go.yml

```diff
@@ -30,3 +30,27 @@ jobs:
 
     - name: Test
       run: go test -v ./...
+
+  check-lint:
+    name: Lint checks
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.17
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: golangci-lint
+      uses: golangci/golangci-lint-action@v2
+      with:
+        # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+        version: v1.42.0
+        working-directory: .
+
+    - name: Check linting
+      run: make lint
```

Comment:
The `golangci/golangci-lint-action@v2` action installs _and_ runs golangci-lint command so it must happen after the code checkout.

I've left the `make lint` step because that's what a developer would run on their machine.